src/server/routes/queueDelete.ts

```diff
@@ -0,0 +1,174 @@
+      options: (status) => ['FREE', 'PREMIUM'].includes(status),
+    },
+  },
+};
+
+router.post(
+  '/',
+  checkSchema(batchDeleteSchema),
+  validate,
+  (req: Request, res: Response) => {
+    const traceId = req.body.traceId ?? nanoid();
+    const savedItemService = new SavedItemDataService({
+      userId: req.body.userId.toString(),
+      dbClient: readClient(),
+      apiId: 'backend',
+    });
+
+    enqueueSavedItemIds(req.body, savedItemService, traceId);
+
+    return res.send({
+      status: 'OK',
+      message: `QueueDelete: Enqueued items for User ID: ${req.body.userId} (traceId='${traceId}')`,
+    });
+  }
+);
+
+/**
+ * Enqueue item IDs for deletions in batches
+ * @param data
+ * @param savedItemService
+ * @param traceId
+ */
+export async function enqueueSavedItemIds(
+  data: Omit<SqsMessage, 'itemIds'>,
+  savedItemService: SavedItemDataService,
+  traceId: string
+): Promise<void> {
+  const { userId, email, status } = data;
+  const limit = config.queueDelete.queryLimit;
+  let offset = 0;
+  const sqsSends: Promise<SendMessageBatchCommandOutput>[] = [];
+  let sqsEntries: SendMessageBatchRequestEntry[] = [];
+
+  const loopCondition = true;
+  while (loopCondition) {
+    const ids = await savedItemService.getSavedItemIds(offset, limit);
+    if (!ids.length) break;
+
+    const chunkedIds = chunk(ids, config.queueDelete.itemIdChunkSize);
+
+    let nextChunk = chunkedIds.next();
+    while (!nextChunk.done) {
+      sqsEntries.push(
+        convertToSqsEntry({
+          userId,
+          email,
+          status,
+          itemIds: nextChunk.value,
+        })
+      );
+
+      if (sqsEntries.length === config.aws.sqs.batchSize) {
+        sqsSends.push(sqsSendBatch(sqsEntries, traceId));
+        sqsEntries = []; // reset
+      }
+
+      nextChunk = chunkedIds.next();
+    }
+
+    offset += offset + limit;
+  }
+
+  // If there's any remaining, send to SQS
+  if (sqsEntries.length) {
+    sqsSends.push(sqsSendBatch(sqsEntries, traceId));
+  }
+
+  try {
+    await Promise.all(sqsSends);
+  } catch (e) {
+    const message = `QueueDelete: Error - Failed to enqueue saved items for userId: ${userId} (traceId='${traceId}')`;
+    Sentry.addBreadcrumb({ message });
+    Sentry.captureException(e);
+    console.log(message);
+  }
+}
+
+/**
+ * Send messages in a batch to SQS
+ * @param entries
+ * @param traceId
+ */
+async function sqsSendBatch(
+  entries: SendMessageBatchRequestEntry[],
+  traceId: string
+): Promise<SendMessageBatchCommandOutput> {
+  const command = new SendMessageBatchCommand({
+    Entries: entries,
+    QueueUrl: config.aws.sqs.listDeleteQueue.url,
+  });
+
+  return sqs.send(command);
+}
+
+/**
+ * Convert a JSON object to an SQS send message entry
+ * @param message
+ */
+function convertToSqsEntry(message: SqsMessage): SendMessageBatchRequestEntry {
+  return {
+    Id: nanoid(),
+    MessageBody: JSON.stringify(message),
+  };
+}
+
+/**
+ * Generate and return a list of a given size
```

Comment:
I think this docstring is wrong -- more like you're yielding chunks of a list vs. generating a list of specific size